velox/functions/sparksql/tests/SparkCastExprTest.cpp

```diff
@@ -230,6 +252,34 @@ TEST_F(SparkCastExprTest, stringToTimestamp) {
   testCast<std::string, Timestamp>("timestamp", input, expected);
 }
 
+TEST_F(SparkCastExprTest, intToTimestamp) {
+  // int64_t
+  testCast(
+      makeNullableFlatVector<int64_t>({
+          0,
+          1727181032,
+          -1727181032,
+          9223372036855,
+          -9223372036856,
+          std::numeric_limits<int64_t>::max(),
+          std::numeric_limits<int64_t>::min(),
+      }),
+      makeNullableFlatVector<Timestamp>({
+          Timestamp(0, 0),
+          Timestamp(1727181032, 0),
+          Timestamp(-1727181032, 0),
+          Timestamp(9223372036854, 775'807'000),
+          Timestamp(-9223372036855, 224'192'000),
+          Timestamp(9223372036854, 775'807'000),
+          Timestamp(-9223372036855, 224'192'000),
+      }));
+
+  // Ensure that other integral types cast to int64 work as well.
+  testIntegralToTimestampCast<int8_t>();
+  testIntegralToTimestampCast<int16_t>();
+  testIntegralToTimestampCast<int32_t>();
```

Comment:
I wonder why we adopt different test methods for int64_t and the others. Is there any special consideration?

Reply:
since other integral types can't test large timestamp values, and since we only need to ensure `cast` smaller integral to `int64_t` works well, so here I adopt different test methods